src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -4574,6 +4574,8 @@ def update_image_galleries(cmd, resource_group_name, gallery_name, gallery, perm
             gallery.soft_delete_policy.is_soft_delete_enabled = soft_delete
         else:
             gallery.soft_delete_policy = {'is_soft_delete_enabled': soft_delete}
+    else:
+        gallery.soft_delete_policy = None
```

Comment:
Because this is a special temporary logic, could we add comments to explain it:
This is a workaround to solve historical legacy issues, send `None` to the service will let service not modify this property. We can delete this logic when the service no longer checks AFEC in the future.

Reply:
Done